### PR TITLE
Removes Flow dependency from the React compiler

### PR DESCRIPTION
**Pull request by trueadm**
Release notes: none

We no longer need Flow with the recent purity changes and this also makes the tests require less dependencies. Other small refactors were made to make way for removal of the Flow code.

### CODE REVIEW

File: src/react/components.js [hunk 1]
```diff
@@ -156,15 +128,17 @@ export function createClassInstance(
...
   // assign props
   Properties.Set(realm, instance, "props", props, true);
   // assign context
   Properties.Set(realm, instance, "context", context, true);
   // enable serialization to support simple instance variables properties
   instance.refuseSerialization = false;
   // return the instance in an abstract object
-  return createAbstractObject(realm, "this", null, instance);
+  let value = createAbstractObject(realm, "this", instance);
+  invariant(value instanceof AbstractObjectValue);
```
Comment (hermanventer) [inline src/react/components.js]: The type annotation of createAbstractObject should already guarantee this.

File: src/intrinsics/prepack/utils.js [hunk 2]
```diff
@@ -111,3 +112,22 @@ export function createAbstract(
     result = AbstractValue.createAbstractConcreteUnion(realm, result, ...additionalValues);
   return result;
 }
+
+export function createAbstractObject(realm: Realm, name: string, template?: ObjectValue): AbstractObjectValue {
```
Comment (hermanventer) [inline src/intrinsics/prepack/utils.js]: We have a rule that all abstract values MUST be created in AbstractValue.

File: src/react/components.js (hunk 1 above)
Comment (trueadm) [reply src/react/components.js]: Flow complains for me :(

File: src/intrinsics/prepack/utils.js (hunk 2 above)
Comment (trueadm) [reply src/intrinsics/prepack/utils.js]: I can move this function out. Thanks for letting me know.

File: src/react/components.js (hunk 1 above)
Comment (hermanventer) [reply src/react/components.js]: Then create an issue for the Flow team and use FlowFixMe to suppress the error and to reference the issue.

**Comment by sophiebits** [comment]
happyjello

**Pull request closed without merging**